hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/client/ScmClient.java

```diff
@@ -128,6 +128,19 @@ void deleteContainer(long containerId, Pipeline pipeline, boolean force)
   List<ContainerInfo> listContainer(long startContainerID,
       int count) throws IOException;
 
+  /**
+   * Lists a range of containers and get their info.
+   *
+   * @param startContainerID start containerID.
+   * @param count count must be {@literal >} 0.
+   *
+   * @return a list of containers capped by max count allowed
+   * in "hdds.container.list.max.count" and total number of containers.
+   * @throws IOException
+   */
+  Pair<List<ContainerInfo>, Long> listContainerWithCount(long startContainerID,
+      int count) throws IOException;
```

Comment:
Please consider creating a wrapper object for the return type, instead of using `Pair`.  Having a custom type allows it to evolve, if necessary.  It also makes the code more readable, by using more specific method names instead of `getLeft`, `getRight`.